### PR TITLE
Make all alt builders produce parallel-enabled compilers

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
We're not quite ready to ship parallel compilers by default, but the alt
builders are not used too much (in theory), so we believe that shipping
a possibly-broken compiler there is not too problematic.

r? @nikomatsakis

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
Going to nominate for infra and compiler teams as I'd like to check that the claims I've made are true:
 * usage is minimal, mostly to deal with LLVM-related regressions (-Zthreads=1 is probably enough to sidestep any bugs in parallel that could impact this)
 * we are comfortable altering them in such a "big" way.

**Comment by pnkfelix** [comment]
Discussed in T-compiler meeting.  We ourselves don't see any problem with doing this as described here.

In particular, @nikomatsakis wanted to stress that we *do* plan to keep LLVM assertions enabled for these builds. This will mean that these builds help flush out parallel bugs, but they will *not* be a good basis for comparative benchmarking.

**Comment by pnkfelix** [comment]
(leaving nominated tag on, but at this point that is directed at T-infra, not T-compiler.)

**Comment by Mark-Simulacrum** [comment]
r? @alexcrichton 

I believe we discussed this in the infra meeting and there weren't any major objections -- could you just make sure this is doing what it's intended and such and approve if you think we're good to go here?

**Comment by alexcrichton** [comment]
Would it be possible to configure these compilers to be built with `-j1` for everything except the codegen backend? I'm basically curious if we can just enable parallelism but not use it.

**Comment by Mark-Simulacrum** [comment]
I _think_ that's "just" `-Zthreads=1` at runtime but I can take a look to both verify that theory and, if so, hard code that as the default (perhaps via a cfg flag or something).

**Comment by alexcrichton** [comment]
Ah ok, great! In that case actually, I think it'd be best to default to `-Zthreads=1` by default? Basically leave everything as-is, but just producing "parallel capable" compilers here.

**Comment by Mark-Simulacrum** [comment]
Alright, I pushed up another commit that makes the default for query parallelism 1, instead of num_cpus as before. This does _not_ affect the codegen parallelism.

File: src/librustc/session/config.rs [hunk 1]
```diff
@@ -1259,7 +1259,7 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
         "prints the LLVM optimization passes being run"),
     ast_json: bool = (false, parse_bool, [UNTRACKED],
         "print the AST as JSON and halt"),
-    threads: Option<usize> = (None, parse_opt_uint, [UNTRACKED],
+    threads: Option<usize> = (Some(1), parse_opt_uint, [UNTRACKED],
```
Comment (alexcrichton) [inline src/librustc/session/config.rs]: It seems a bit odd to have this be `Option<usize>` when the default isn't `None`, could this instead change `threads_from_count`, and a comment could be included why it's defaulted to one?

Comment (Mark-Simulacrum) [reply src/librustc/session/config.rs]: Sure, yeah.

**Comment by Mark-Simulacrum** [comment]
Was a bit overeager pushing, there's a new commit up (no significant differences)

**Pull request merged**